### PR TITLE
[JSON-Logic] Part 0: Allow consts to support single values only

### PR DESCRIPTION
**Pull request by brennj**
Part 1 of merging JSON Logic schemas in, but this is really its own standalone piece of work.

Today if we have a `const` value in a schema, it doesn't take into account validations to only allow that value.

With this PR, it adds that functionality to cause an error, also where you can define the custom error using `x-jsf-errorMessage.const`.

Please see the tests for sample schemas on how to make this work.

### CODE REVIEW

**Review by johnstonbl01 — DISMISSED** [review]

File: src/yupSchema.js [hunk 1]
```diff
@@ -304,6 +304,19 @@ export function buildYupSchema(field, config) {
     );
   }
 
+  function withConst(yupSchema) {
+    return yupSchema.test(
+      'isConst',
+      errorMessage.const ??
```
[the remark this answers is not in the document]
Comment (sandrina-p) [reply src/yupSchema.js:310]: We have just [one sentence](https://json-schema-form.vercel.app/?path=/docs/api-reference-json-schema--custom-keywords#x-jsf-errormessage) about that it the docs. Probably a table or diagram could explain it better.

File: src/tests/const.test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import { createHeadlessForm } from '../createHeadlessForm';
+
+it('Should work for number', () => {
```
Comment (sandrina-p) [inline src/tests/const.test.js:3]: question: Is this the moment we start to break our tests into smaller files? If so this should be a `describe('validations: const', ())` with all the `it` inside it.

File: src/tests/const.test.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        ten_only: { type: 'number', const: 10 },
+      },
+    },
+    { strictInputType: false }
+  );
+  expect(handleValidation({}).formErrors).toEqual(undefined);
+  expect(handleValidation({ ten_only: 1 }).formErrors).toEqual({
+    ten_only: 'The only accepted value is 10.',
+  });
+  expect(handleValidation({ ten_only: 10 }).formErrors).toEqual(undefined);
```
Comment (sandrina-p) [inline src/tests/const.test.js:16]: todo: It's missing to test the `ten_only: null` to cover the "bug" `@BUG RMT-518` (check other comments in the `createHeadlessForm.test.js`)

```js
  // null is also considered valid until we fix @BUG RMT-518
  // Expectation: To fail with error "The only accepted value is 10."
  expect(handleValidation({ ten_only: null }).formErrors).toBeUndefined();
```

File: src/tests/const.test.js (hunk 2 above)
Comment (brennj) [reply src/tests/const.test.js:3]: yes it is, no more massive test files 😅 , will fix this.

File: src/tests/const.test.js (hunk 3 above)
Comment (brennj) [reply src/tests/const.test.js]: Added!

**Comment by brennj** [comment]
@sandrina-p are you the admin on this repo? Can we change it so that new commits dismiss old reviews?

if you or @johnstonbl01 can give this a "reapproval" i think we can merge now, green internally now!

File: src/tests/const.test.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import { createHeadlessForm } from '../createHeadlessForm';
```
Comment (sandrina-p) [inline src/tests/const.test.js:1]: > @sandrina-p are you the admin on this repo? Can we change it so that new commits dismiss old reviews?

> if you or @johnstonbl01 can give this a "reapproval" i think we can merge now, green internally now!


I hate github for not allowing threads in "standalone" comments, so I'm creating this new comment:

Comment (sandrina-p) [reply src/tests/const.test.js:1]: Yes, I was the only admin, I changed that now, pls double-check. I also added a branch rule protection to `main` to require re-approvals after commits.

Comment (brennj) [reply src/tests/const.test.js:1]: Looks good, thank you!
